runtime/permissions/lib.rs

```diff
@@ -1330,22 +1321,38 @@ impl UnaryPermission<RunDescriptor> {
 
   pub fn check(
     &mut self,
-    cmd: &str,
+    cmd: &Path,
     api_name: Option<&str>,
   ) -> Result<(), AnyError> {
+    debug_assert!(cmd.is_absolute());
     skip_check_if_is_permission_fully_granted!(self);
     self.check_desc(
-      Some(&RunDescriptor::from(cmd.to_string())),
+      Some(&RunDescriptor::Path(cmd.to_path_buf())),
       false,
       api_name,
-      || Some(format!("\"{}\"", cmd)),
+      || Some(format!("\"{}\"", cmd.display())),
     )
   }
 
   pub fn check_all(&mut self, api_name: Option<&str>) -> Result<(), AnyError> {
     skip_check_if_is_permission_fully_granted!(self);
     self.check_desc(None, false, api_name, || None)
   }
+
+  /// Queries without prompting
+  pub fn query_all(&mut self, api_name: Option<&str>) -> bool {
+    if self.is_allow_all() {
+      return true;
+    }
+    let (result, _prompted, _is_allow_all) =
+      self.query_desc(None, AllowPartial::TreatAsDenied).check2(
+        RunDescriptor::flag_name(),
+        api_name,
+        || None,
+        /* no prompt */ false,
```

Comment:
This should be true

Reply:
Turns out this should be `/* prompt */` instead and there is a test for it